kratos/input_output/vtk_output.cpp

```diff
@@ -159,16 +160,26 @@ std::string VtkOutput::GetOutputFileName(const ModelPart& rModelPart, const bool
         ss << std::setprecision(mDefaultPrecision) << std::setfill('0')
            << rModelPart.GetProcessInfo()[TIME];
         label = ss.str();
+    } else if(output_control == "nl_iteration_number") {
+        ss << std::setprecision(mDefaultPrecision) << std::setfill('0')
+           << rModelPart.GetProcessInfo()[NL_ITERATION_NUMBER];
+        label = ss.str();
     } else {
         KRATOS_ERROR << "Option for \"output_control_type\": " << output_control
             <<" not recognised!\nPossible output_control_type options "
             << "are: \"step\", \"time\"" << std::endl;
     }
 
     // Putting everything together
+    const std::string& r_folder_name = mOutputSettings["folder_name"].GetString();
+    const bool directory_exists = OSUtilities::IsDirExist(r_folder_name);
```

Comment:
I am not overly happy with this (because so far we always did this in python) but I guess this is the only way if we are moving towards using this also in C (I assume you are doing this aren't you)?

Also if we have OS-independent solutions now I think it is ok

Reply:
I am moving between Linux and Windows, now in Windows works, and in Linux looks like it fails. Second iteration